Guide/src/reference/architecture/openhcl/processes.md

```diff
@@ -0,0 +1,121 @@
+- **Dispatch Loop:** Runs a minimal loop waiting for commands from the host or the main kernel.
+- **On-Demand Conversion:** Can be converted to a full Linux CPU if required.
+
+## Init Process (`underhill_init`)
+
+`underhill_init` is the first userspace process (PID 1) started by the Linux kernel. It acts as the system service manager for the paravisor environment.
+
+**Source code:** [openhcl/underhill_init](https://github.com/microsoft/openvmm/tree/main/openhcl/underhill_init) | **Docs:** [underhill_init rustdoc](https://openvmm.dev/rustdoc/linux/underhill_init/index.html)
+
+**Key Responsibilities:**
+
+- **System Setup:** Mounts necessary filesystems (e.g., `/proc`, `/sys`, `/dev`).
+- **Environment Preparation:** Sets up the execution environment for the paravisor.
+- **Process Launch:** `exec`s the main paravisor process (`openvmm_hcl`).
+
+## Paravisor (`openvmm_hcl`)
+
+`openvmm_hcl` is the central management process of the OpenHCL paravisor. It runs in userspace and orchestrates the virtualization services.
+
+**Source code:** [openhcl/openvmm_hcl](https://github.com/microsoft/openvmm/tree/main/openhcl/openvmm_hcl) | **Docs:** [openvmm_hcl rustdoc](https://openvmm.dev/rustdoc/linux/openvmm_hcl/index.html)
+
+**Key Responsibilities:**
+
+- **Policy & Management:** Manages the lifecycle of the VM and enforces security policies.
+- **Host Communication:** Interfaces with the host VMM to receive commands and report status.
+- **Servicing:** Orchestrates save and restore operations (VTL2 servicing).
+- **Worker Management:** Spawns and manages the VM worker process.
+
+## VM Worker (`underhill_vm`)
+
+The VM worker process (`underhill_vm`) is responsible for the high-performance data path of the virtual machine. It is spawned by `openvmm_hcl`.
+
+**Source code:** [openhcl/underhill_core](https://github.com/microsoft/openvmm/tree/main/openhcl/underhill_core) | **Docs:** [underhill_core rustdoc](https://openvmm.dev/rustdoc/linux/underhill_core/index.html)
+
+**Key Responsibilities:**
+
+- **VP Loop:** Runs the virtual processor loop, handling VM exits.
+- **Device Emulation:** Coordinates device emulation for the guest VM. Some devices run in-process while others run in separate device worker processes for isolation.
+- **I/O Processing:** Handles high-speed I/O operations.
+
+## Diagnostics Server (`diag_server`)
+
+The diagnostics server provides an interface for debugging and monitoring the OpenHCL environment.
+
+**Source code:** [openhcl/diag_server](https://github.com/microsoft/openvmm/tree/main/openhcl/diag_server) | **Docs:** [diag_server rustdoc](https://openvmm.dev/rustdoc/linux/diag_server/index.html)
+
+**Key Responsibilities:**
+
+- **External Interface:** Listens on a VSOCK port for diagnostic connections.
+- **Command Handling:** Processes diagnostic commands and queries.
+- **Log Retrieval:** Provides access to system logs.
+
+## Profiler Worker (`profiler_worker`)
```

Comment:
I think this is only for azure profiler, which will only show up in internal builds, right? Probably not worth including here.